### ISSUE TITLE
Compile fails on ESP8266

### ISSUE THREAD

**Issue by rajil** [DrGFreeman/SharpDistSensor#12]
Hi,

I tried to compile it with nodemcu but it failed,

```
Build options changed, rebuilding all
/home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp: In constructor 'MedianFilter::MedianFilter(byte, int)':
/home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:47:28: error: no matching function for call to 'max(const byte&, int)'
  medFilterWin = max(size, 3);   // number of samples in sliding median filter window - usually odd #

                            ^
/home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:47:28: note: candidates are:
In file included from /home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/algorithm:62:0,
                 from /home/user/.arduino15/packages/esp8266/hardware/esp8266/2.4.1/cores/esp8266/Arduino.h:254,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.h:36,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:40:
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algo.h:4236:5: note: template<class _Tp, class _Compare> _Tp std::max(std::initializer_list<_Tp>, _Compare)
     max(initializer_list<_Tp> __l, _Compare __comp)
     ^
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algo.h:4236:5: note:   template argument deduction/substitution failed:
/home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:47:28: note:   mismatched types 'std::initializer_list<_Tp>' and 'unsigned char'
  medFilterWin = max(size, 3);   // number of samples in sliding median filter window - usually odd #

                            ^
In file included from /home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/algorithm:62:0,
                 from /home/user/.arduino15/packages/esp8266/hardware/esp8266/2.4.1/cores/esp8266/Arduino.h:254,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.h:36,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:40:
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algo.h:4231:5: note: template<class _Tp> _Tp std::max(std::initializer_list<_Tp>)
     max(initializer_list<_Tp> __l)
     ^
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algo.h:4231:5: note:   template argument deduction/substitution failed:
/home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:47:28: note:   mismatched types 'std::initializer_list<_Tp>' and 'unsigned char'
  medFilterWin = max(size, 3);   // number of samples in sliding median filter window - usually odd #

                            ^
In file included from /home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/algorithm:61:0,
                 from /home/user/.arduino15/packages/esp8266/hardware/esp8266/2.4.1/cores/esp8266/Arduino.h:254,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.h:36,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:40:
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algobase.h:260:5: note: template<class _Tp, class _Compare> const _Tp& std::max(const _Tp&, const _Tp&, _Compare)
     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
     ^
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algobase.h:260:5: note:   template argument deduction/substitution failed:
/home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:47:28: note:   deduced conflicting types for parameter 'const _Tp' ('unsigned char' and 'int')
  medFilterWin = max(size, 3);   // number of samples in sliding median filter window - usually odd #

                            ^
In file included from /home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/algorithm:61:0,
                 from /home/user/.arduino15/packages/esp8266/hardware/esp8266/2.4.1/cores/esp8266/Arduino.h:254,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.h:36,
                 from /home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:40:
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algobase.h:216:5: note: template<class _Tp> const _Tp& std::max(const _Tp&, const _Tp&)
     max(const _Tp& __a, const _Tp& __b)
     ^
/home/user/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algobase.h:216:5: note:   template argument deduction/substitution failed:
/home/user/Arduino/libraries/SharpDistSensor/MedianFilter.cpp:47:28: note:   deduced conflicting types for parameter 'const _Tp' ('unsigned char' and 'int')
  medFilterWin = max(size, 3);   // number of samples in sliding median filter window - usually odd #

                            ^
exit status 1
Error compiling for board NodeMCU 1.0 (ESP-12E Module).
 
```

**Comment by DrGFreeman** [comment]
Hello @rajil, thanks for raising the issue.

The issue is in the MedianFilter constructor. I have integrated the [daPhoosa/MedianFilter](https://github.com/daPhoosa/MedianFilter) library directly into mine. Apparently, the same issue was raised on that library [here](https://github.com/daPhoosa/MedianFilter/issues/2) and fixed. When I have a minute, I will update the MedianFilter files from the original library into mine. This should solve the issue.

In the meantime, you can copy the latest MedianFilter.h and MedianFilter.cpp from the original  [daPhoosa/MedianFilter](https://github.com/daPhoosa/MedianFilter) library into your SharpDistSensor library folder and it should work.

Best regards,
Julien

**Comment by DrGFreeman** [comment]
Fixed [version 1.4.1](https://github.com/DrGFreeman/SharpDistSensor/releases/tag/1.4.1) is now released and should be available in the Arduino Library Manager soon.
